src/gasman.c

```diff
@@ -1943,6 +1960,11 @@ UInt CollectBags (
 
     /* * * * * * * * * * * * * * * check phase * * * * * * * * * * * * * * */
 
+    // Check if this allocation would even fit into memory
+    if(((size_t)-1) - (size_t)(sizeof(BagHeader)+size) < (size_t)AllocBags) {
```

Comment:
missing space after `if`. Also `((size_t)-1)`  should be `SIZE_MAX` (from ISO C).

But perhaps the more idiomatic way would be to add a check after setting `stopBags`:
```C
// check for an overflow
if (stopBags < AllocBags)
  SyAbortBags("gap: cannot extend the workspace any more!!");
```

(I assume you want to abort, not `return 0`?

Reply:
Pointer overflow isn't defined. gcc will (for example) optimise `i < i + 1` away to true for pointer `i`